cabal-install/Distribution/Client/CmdRepl.hs

```diff
@@ -335,16 +338,23 @@ data OriginalComponentInfo = OriginalComponentInfo
   }
   deriving (Show)
 
-withProject :: ProjectConfig -> Verbosity -> [String] -> IO (ProjectBaseContext, [TargetSelector], IO ())
+-- | Tracks what type of GHCi instance we're creating.
+data ReplType = ProjectRepl 
+              | GlobalRepl FilePath -- ^ The 'FilePath' argument is the correct CWD.
```

Comment:
As I read more closely, that argument seems to be an argument to a `-ghci-script`.